pkg/gofr/datasource/pubsub/google/google_test.go

```diff
@@ -50,6 +50,24 @@ func TestGoogleClient_New_Error(t *testing.T) {
 	assert.Contains(t, out, "could not configure google pubsub")
 }
 
+func TestGoogleClient_Close(t *testing.T) {
+	var (
+		g *googleClient
+	)
+
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	logger := logging.NewMockLogger(logging.ERROR)
+	g = New(Config{}, logger, NewMockMetrics(ctrl))
+
+	assert.Nil(t, g)
```

Comment:
I don't get the `assert.Nil` , and then a `g.Close()` on next lines. It would lead to a nil pointer exception no?